### PR TITLE
Case 20534: Fix light crash

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/20534/Backtrace-graphics-Light-getDirection-272d78230b31e2f51d7ec2f848114e10c2dabe0bdfd43f57d7239f7984c5f678-58-Crashes-in-8-days

Test plan:
- Make sure shadows and the keylight in a zone still work.
- I don't know what causes this crash.  It sounds like it could be related to switching between two domains via GOTO.  The best plan might be to merge this and then confirm that the crash stops appearing in backtrace in 78.

### CODE REVIEW

File: libraries/render-utils/src/RenderShadowTask.cpp [hunk 1]
```diff
@@ -356,92 +355,90 @@ void RenderShadowSetup::run(const render::RenderContextPointer& renderContext, c
...
+    args->pushViewFrustum(*_coarseShadowFrustum);
+
+    args->_renderMode = RenderArgs::SHADOW_RENDER_MODE;
+
+    // We want for the octree query enough resolution to catch the details in the lowest cascade. So compute
+    // the desired resolution for the first cascade frustum and extrapolate it to the coarse frustum.
+    glm::ivec2 queryResolution = firstCascade.framebuffer->getSize();
+    queryResolution.x = int(queryResolution.x * _coarseShadowFrustum->getWidth() / firstCascadeFrustum->getWidth());
+    queryResolution.y = int(queryResolution.y * _coarseShadowFrustum->getHeight() / firstCascadeFrustum->getHeight());
+    output.edit1() = queryResolution;
```
Comment (SamGondelman) [inline libraries/render-utils/src/RenderShadowTask.cpp]: The above is just indentation because I got rid of a duplicate check that the light is non-null.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16321/HighFidelity-Beta-PR14671-928ea55.apk

File: libraries/render-utils/src/RenderShadowTask.cpp [hunk 2]
```diff
@@ -61,12 +61,11 @@ void RenderShadowTask::build(JobModel& task, const render::Varying& input, rende
     // FIXME: calling this here before the zones/lights are drawn during the deferred/forward passes means we're actually using the frames from the previous draw
     // Fetch the current frame stacks from all the stages
     // Starting with the Light Frame  genreated in previous tasks
```
Comment (sethalves) [inline libraries/render-utils/src/RenderShadowTask.cpp]: is this comment still correct?

Comment (SamGondelman) [reply libraries/render-utils/src/RenderShadowTask.cpp]: Although the exact wording of the comment is wrong (even in master right now), the general idea that we're using the wrong frame's stages is correct, I believe, so I left it for now.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16321/HighFidelity-Beta-PR14671-928ea55.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16321/HighFidelity-Beta-Interface-PR14671-928ea55.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16321/HighFidelity-Beta-PR14671-928ea55.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16321/HighFidelity-Beta-Interface-PR14671-928ea55.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass
Spent a good while switching domains and messing with various zones that had lighting and shadows enabled and disabled. No crash 🎉

**Pull request merged**